Makefile.common

```diff
@@ -108,7 +108,15 @@ CXXFLAGS += -fmacro-prefix-map=$(dir $(TOPDIR))=
 # garbage-collect unused code/data
 # NOTE: exported to ports also as it reduces binaries size greatly
 CFLAGS += -ffunction-sections -fdata-sections
-LDFLAGS += $(LDFLAGS_PREFIX)--gc-sections
+ifeq ($(APPLE_LD), 1)
+  LDFLAGS += $(LDFLAGS_PREFIX)-dead_strip
+  LDFLAG1 = $(LDFLAGS_PREFIX)-force_load
+  LDFLAG2 = $(LDFLAGS_PREFIX)
+else
+  LDFLAGS += $(LDFLAGS_PREFIX)--gc-sections
+  LDFLAG1 = $(LDFLAGS_PREFIX)--whole-archive
+  LDFLAG2 = $(LDFLAGS_PREFIX)--no-whole-archive
+endif
```

Comment:
I would go for default parameters here (to highlight the most common code path) and overwrite it in `host.mk`, eg in `Makefile.common`:

```make
LDFLAGS_GC_SECTIONS ?= $(LDFLAGS_PREFIX)--gc-sections
LDFLAGS_WHOLE_ARCHIVE_BEGIN ?= $(LDFLAGS_PREFIX)--whole-archive
LDFLAGS_WHOLE_ARCHIVE_END ?= $(LDFLAGS_PREFIX)--no-whole-archive

LDFLAGS += $(LDFLAGS_GC_SECTIONS)
```

in `host.mk`:
```make

# [...] if apple linker
LDFLAGS_GC_SECTIONS := $(LDFLAGS_PREFIX)-dead_strip
LDFLAGS_WHOLE_ARCHIVE_BEGIN := $(LDFLAGS_PREFIX)-all_load
LDFLAGS_WHOLE_ARCHIVE_END := 
# (above explicitly set to empty, so ?= won't assign default value)

```